rust/worker/src/execution/orchestration/compact.rs

```diff
@@ -271,6 +277,21 @@ impl CompactOrchestrator {
         let input = PartitionInput::new(records, self.max_partition_size);
         let task = wrap(operator, input, ctx.receiver());
         self.send(task, ctx).await;
+
+        let segments = self.get_all_segments().await.unwrap();
```

Comment:
(Not putting this in the `initial_tasks()` method from the Orchestrator trait as that would require making `initial_tasks()` async and my understanding is that it's intended to be a static list of tasks. A cleaner solution would perhaps be adding a `GetSegment` operator and dispatching that from `initial_tasks()`.)